### PR TITLE
Add section about mime types

### PR DESCRIPTION
**Pull request by dhh1128**
Signed-off-by: Daniel Hardman <daniel.hardman@gmail.com>

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

**Comment by dhh1128** [comment]
Discussed on community call, 26 Oct 2020. Decided we'd add a section about signed messages. I'll come back with an update about it, shortly.

Also decided we need a way to evaluate variations on the naming scheme for the types. I've created this spreadsheet where we can compare alternatives: https://docs.google.com/spreadsheets/d/1VZz8J2Uz4nab-SY4pvhKd1_eipoEo9kGiqU5iaMYVsY/edit?usp=sharing

File: message_structure.md [hunk 1]
```diff
@@ -39,6 +39,40 @@ Aligning with [RFC 6648](https://tools.ietf.org/html/rfc6648.html), DIDComm expl
...
+
+When higher-level protocols are built atop DIDComm, applications remove the protective envelope and process the plaintext that's inside. Specs for such protocols typically document message structure and provide examples in this format; the encrypted envelope is assumed but ignored as a low-level detail.
+
+Applications running one or more DIDComm-based protocols may wish to define their own media types for the protocols they support. Absent such customization, the preferred media type for a generic DIDComm plaintext message SHOULD be `application/didcomm-plaintext+json`. This correctly conveys the fact that code handling such content functions at a different level from DIDComm's internals, and that generic JSON tools and actions are likely to be a helpful fallback in processing the content.
```
Comment (tmarkovski) [inline message_structure.md:64]: I would like to propose the use of the following notation for the mime types
application/didcomm.plaintext+json
application/didcomm.encrypted+json
application/didcomm.signed+json

I'm less opinionated on the language of `plaintext`, `encrypted` and `signed`.

Based on mime type notes at https://en.wikipedia.org/wiki/Media_type

**Comment by tmarkovski** [comment]
Added some more options to the doc to throw ideas.

**Comment by TelegramSam** [comment]
application/didcomm-encrypted+json | application/didcomm-signed+json | application/didcomm-plain+json

**Pull request merged**